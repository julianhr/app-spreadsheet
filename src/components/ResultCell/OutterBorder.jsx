/* @jsx jsx */
import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'
import { jsx, css } from '@emotion/core' // eslint-disable-line
import { connect } from 'react-redux'

import { DEFAULT_COL_WIDTH } from '~/library/constants'


const Root = styled.div`
  cursor: cell;
  overflow: hidden;
  font-size: 13px;
  border-right: 1px solid ${props => props.theme.colors.table.border};
  border-bottom: 1px solid ${props => props.theme.colors.table.border};
  outline: none;
`

export function OutterBorder({ width, height, onClick, onDoubleClick, children }) {
  return (
    <Root
      data-cell='border'
      onClick={onClick}
      onDoubleClick={onDoubleClick}
      css={{
        width,
        height: height && height
      }}
    >
      {children}
    </Root>
  )
}

OutterBorder.propTypes = {
  onClick: PropTypes.func.isRequired,
  onDoubleClick: PropTypes.func.isRequired,
  location: PropTypes.string.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
}

OutterBorder.defaultProps = {
  width: DEFAULT_COL_WIDTH,
}

function mapStateToProps(state, ownProps) {
  const [colLabel, rowLabel] = ownProps.location.split('-')
  const width = state.tableMeta.colWidths[colLabel]
  const height = state.tableMeta.rowHeights[rowLabel]
  return { width, height }
}

export default connect(mapStateToProps)(OutterBorder)
