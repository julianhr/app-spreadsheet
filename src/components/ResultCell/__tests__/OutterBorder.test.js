import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import emotionSerializer from 'jest-emotion'

import MockApp from '~/__tests__/__mocks__/MockApp'
import { OutterBorder } from '../OutterBorder'


const requiredProps = {
  location: 'A-3',
  onClick: jest.fn(),
  onDoubleClick: jest.fn(),
}

const testProps = {
  ...requiredProps,
  width: 100,
  height: 30,
}

const renderApp = (props) => {
  return render(
    <MockApp>
      <OutterBorder {...props}>
        Test
      </OutterBorder>
    </MockApp>
  )
}

describe('OutterBorder', () => {
  afterEach(cleanup)

  test('required props', () => {
    Object.keys(requiredProps).forEach(prop => {
      const props = { ...requiredProps }
      delete props[prop]
      expect(() => renderApp(props)).toThrow()
    })
  })

  test('default width and height', () => {
    renderApp(requiredProps)
    expect.addSnapshotSerializer(emotionSerializer)
    const el = document.querySelector('[data-cell="border"]')
    expect(el).toMatchSnapshot()
  })

  test('specified width and height', () => {
    renderApp(testProps)
    expect.addSnapshotSerializer(emotionSerializer)
    const el = document.querySelector('[data-cell="border"]')
    expect(el).toMatchSnapshot()
  })
})
