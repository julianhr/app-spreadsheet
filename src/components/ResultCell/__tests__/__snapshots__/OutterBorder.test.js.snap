// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutterBorder default width and height 1`] = `
.emotion-0 {
  cursor: cell;
  overflow: hidden;
  font-size: 13px;
  border-right: 1px solid #dfdfdf;
  border-bottom: 1px solid #dfdfdf;
  outline: none;
  width: 140px;
}

<div
  class="emotion-0 emotion-1"
  data-cell="border"
>
  Test
</div>
`;

exports[`OutterBorder specified width and height 1`] = `
.emotion-0 {
  cursor: cell;
  overflow: hidden;
  font-size: 13px;
  border-right: 1px solid #dfdfdf;
  border-bottom: 1px solid #dfdfdf;
  outline: none;
  width: 100px;
  height: 30px;
}

<div
  class="emotion-0 emotion-1"
  data-cell="border"
>
  Test
</div>
`;
